core/include/gnuradio-4.0/CircularBuffer.hpp

```diff
@@ -436,104 +448,69 @@ class CircularBuffer {
                 return PublishableOutputRange<U, policy>(this);
             }
 
-            try {
-                const auto        sequence = _claimStrategy->next(*_buffer->_read_indices, nSamples); // alt: try_next
-                const std::size_t index    = (static_cast<std::size_t>(sequence) + _size - nSamples) % _size;
-                return PublishableOutputRange<U, policy>(this, index, sequence, nSamples);
-            } catch (const NoCapacityException&) {
+            const auto sequence = _buffer->_claimStrategy.tryNext(nSamples);
```

Comment:
name the type `auto` explicitly to remove the ambiguity between `std::optional` 'value()' and the one from `Sequence`. 

N.B. `auto` should be only used if the type can be inferred from the right-hand side and/or is a large (less important) templated type (e.g. templated return type).